### PR TITLE
Some minor fix

### PR DESCRIPTION
**Pull request by weihuoya**
`num_bytes + alignment` doesn't make sense.
remove unused variate `m_dummy_texture`.
adjust `render_source_rc` , not `xfb_rect`.

### CODE REVIEW

**Comment by JMC47** [comment]
Is there any issues/games that this affects where I can verify the change?  Thanks :)

File: Source/Core/VideoBackends/Vulkan/StreamBuffer.cpp [hunk 1]
```diff
@@ -131,7 +131,7 @@ bool StreamBuffer::AllocateBuffer()
 
 bool StreamBuffer::ReserveMemory(u32 num_bytes, u32 alignment)
 {
-  const u32 required_bytes = num_bytes + alignment;
+  const u32 required_bytes = Common::AlignUp(num_bytes, alignment);
```
Comment (stenzek) [inline Source/Core/VideoBackends/Vulkan/StreamBuffer.cpp]: Are you sure this is correct? The alignment padding here isn't dependent on the new content size, it is the maximum amount of padding which can be added to the current offset to align it.

**Comment by weihuoya** [comment]
> Is there any issues/games that this affects where I can verify the change? Thanks :)

a few game may fall into `PanicAlert`, don't remember clearly.
```
if (required_bytes > m_size)
  {
    PanicAlert("Attempting to allocate %u bytes from a %u byte stream buffer",
               static_cast<uint32_t>(num_bytes), static_cast<uint32_t>(m_size));

    return false;
  }
```

**Comment by Helios747** [comment]
@stenzek does this look ok to you?

**Pull request closed without merging**